terraform/groups/ecs-service/locals.tf

```diff
@@ -2,81 +2,66 @@
-    { "name": "ACCOUNT_URL", "valueFrom": "${local.service_secrets_arn_map.account_url}" },
-    { "name": "CACHE_SERVER", "valueFrom": "${local.service_secrets_arn_map.cache_server}" },
-    { "name": "OAUTH2_CLIENT_ID", "valueFrom": "${local.service_secrets_arn_map.oauth2_client_id}" },
-    { "name": "OAUTH2_CLIENT_SECRET", "valueFrom": "${local.service_secrets_arn_map.oauth2_client_secret}" },
-    { "name": "EXTENSIONS_API_URL", "valueFrom": "${local.service_secrets_arn_map.internal_api_url}" },
-    { "name": "EXTENSIONS_PROCESSOR_API_URL", "valueFrom": "${local.service_secrets_arn_map.internal_api_url}" }
-  ]
+  service_secret_list = flatten([for key, value in local.service_secrets_arn_map :
+    { "name" = upper(key), "valueFrom" = value }
+  ])
 
-  task_environment = [
-    { "name": "NODE_PORT", "value": "${local.container_port}" },
-    { "name": "API_LOCAL_URL", "value": "${var.api_local_url}"},  
-    { "name": "CDN_HOST", "value": "${var.cdn_host}"},
-    { "name": "CHS_URL", "value": "${var.chs_url}"},
-    { "name": "COOKIE_DOMAIN", "value": "${var.cookie_domain}"},
-    { "name": "COOKIE_EXPIRATION_IN_SECONDS", "value": "${var.cookie_expiration_in_seconds}"},
-    { "name": "COOKIE_NAME", "value": "${var.cookie_name}"},
-    { "name": "COOKIE_SECURE_FLAG", "value": "${var.cookie_secure_flag}"},
-    { "name": "DEFAULT_SESSION_EXPIRATION", "value": "${var.default_session_expiration}"},
-    { "name": "FEATURE_REQUEST_COUNT", "value": "${var.feature_request_count}"},
-    { "name": "HUMAN_LOG", "value": "${var.human_log}"},
-    { "name": "LOG_LEVEL", "value": "${var.log_level}"},
-    { "name": "MAXIMUM_EXTENSION_REQUESTS_PER_DAY", "value": "${var.maximum_extension_requests_per_day}"},
-    { "name": "MAX_EXTENSION_PERIOD_IN_MONTHS", "value": "${var.max_extension_period_in_months}"},
-    { "name": "MAX_FILE_SIZE_BYTES", "value": "${var.max_file_size_bytes}"},
-    { "name": "PERMISSION_NAME_DOWNLOAD", "value": "${var.permission_name_download}"},
-    { "name": "PERMISSION_NAME_VIEW", "value": "${var.permission_name_view}"},
-    { "name": "PIWIK_SITE_ID", "value": "${var.piwik_site_id}"},
-    { "name": "PIWIK_URL", "value": "${var.piwik_url}"},
-    { "name": "SESSION_CREATE", "value": "${var.session_create}"},
-    { "name": "SHOW_SERVICE_UNAVAILABLE_PAGE", "value": "${var.show_service_unavailable_page}"},
-    { "name": "TOO_SOON_DAYS_BEFORE_DUE_DATE", "value": "${var.too_soon_days_before_due_date}"},
+  ssm_service_version_map = [
+    for sec in module.secrets.secrets : {
+      name = "${replace(upper(local.service_name), "-", "_")}_${var.ssm_version_prefix}${replace(upper(basename(sec.name)), "-", "_")}", value = sec.version
+    }
   ]
+
+  task_secrets = concat(local.global_secret_list,local.service_secret_list,[])
+
+  task_environment = concat(local.ssm_global_version_map,local.ssm_service_version_map,[
+    { "name" : "EXTENSIONS_WEB_VERSION", "value" : "${var.extensions_web_version}" }
```

Comment:
Is this variable to be passed to the Application? There is a variable defined. Not having a default value for it.

Reply:
Usually this is variable used when running the terraform only.
- task: set-terraform-variables
            $(cat github-release/tag)
            
Not sure if the web application needs this value?

Reply:
it is needed ```          cat <<EOF >> terraform-variables/variables.tfvars
          docker_registry=((secrets.shared-services-eu-west-2-docker-registry))
          docker_repo=extensions-web
          extensions_web_version=$(cat github-release/tag)```